File: java/111424_B4_29807.java
/**
 * https://www.acmicpc.net/problem/29807
 */
import java.io.*;
import java.util.*;

class Main {
	public static void main(String[] args) throws IOException {
		try(var br = new BufferedReader(new InputStreamReader(System.in))) {
			Integer.parseInt(br.readLine());
			StringTokenizer st = new StringTokenizer(br.readLine());
			int hangle = st.hasMoreTokens() ? Integer.parseInt(st.nextToken()) : 0;
			int math = st.hasMoreTokens() ? Integer.parseInt(st.nextToken()) : 0;
			int english = st.hasMoreTokens() ? Integer.parseInt(st.nextToken()) : 0;
			int tamgu = st.hasMoreTokens() ? Integer.parseInt(st.nextToken()) : 0;
			int daini = st.hasMoreTokens() ? Integer.parseInt(st.nextToken()) : 0;
			
			int gakuban = 0;
			gakuban += hangle > english ? (hangle - english) * 508 : (english - hangle) * 108;
			gakuban += math > tamgu ? (math - tamgu) * 212 : (tamgu - math) * 305;
			gakuban += daini > 0 ? daini * 707 : 0;
			gakuban *= 4763;
			
			System.out.println(gakuban);
		}
	}
}

File: php/111424_B4_29807.php
/**
 * https://www.acmicpc.net/problem/29807
 */
<?php
	fgets(STDIN);
	$tokens = explode(" ", trim(fgets(STDIN)));
	
	$hangle = isset($tokens[0]) ? (int)$tokens[0] : 0;
	$math = isset($tokens[1]) ? (int)$tokens[1] : 0;
	$english = isset($tokens[2]) ? (int)$tokens[2] : 0;
	$tamgu = isset($tokens[3]) ? (int)$tokens[3] : 0;
	$daini = isset($tokens[4]) ? (int)$tokens[4] : 0;
	
	$gakuban = 0;
	$gakuban += $hangle > $english ? ($hangle - $english) * 508 : ($english - $hangle) * 108;
	$gakuban += $math > $tamgu ? ($math - $tamgu) * 212 : ($tamgu - $math) * 305;
	$gakuban += $daini > 0 ? $daini * 707 : 0;
	$gakuban *= 4763;
	
	echo $gakuban . PHP_EOL;
?>

